libs/ui/src/components/NeoAudioPlayer/NeoAudioPlayer.vue

```diff
@@ -0,0 +1,203 @@
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { useEventListener, useMediaControls } from '@vueuse/core'
+import { NeoButton, NeoIcon } from '@kodadot1/brick'
+import ProgressBar from './ProgressBar/ProgressBar.vue'
+import { getRandomValues } from '@/components/unique/utils'
+
+const props = defineProps<{
+  src?: string
+}>()
+
+const { eventBus: playerEventBus, unsubscribe: unsubscribePlayerEventBus } =
+  usePlayerEventBus()
+
+const player = ref()
+const audio = ref()
+const loading = ref(false)
+const canStartPlaying = ref(false)
+const id = ref(getRandomValues(1)[0])
+
+const actionStack = ref<
+  { promise: Promise<void>; reject: (reason?: any) => void }[]
+>([])
+
+const { playing, currentTime, duration, muted } = useMediaControls(audio, {
+  src: props.src,
+})
+
+const playDisabled = computed(() => !canStartPlaying.value || loading.value)
+const canPause = computed(() => !audio.value.paused || playing.value)
+const canPlay = computed(() => audio.value.paused || !playing.value)
+
+const formattedDuration = computed(() => {
+  const time = new Date(currentTime.value * 1000).toISOString().slice(14, 19)
+  return time.startsWith('0') ? time.slice(1, 5) : time
+})
+
+const togglePlay = async () => {
+  if (playing.value) {
+    await pause()
+  } else {
+    await play()
+  }
+}
+
+const flushPreviouseActions = async () => {
+  try {
+    await actionStack.value.map((action) => action.reject())
+    actionStack.value = []
+  } catch (error) {}
+}
+
+const pushToActionStack = ({ promise, reject }) => {
+  actionStack.value.push({
+    promise,
+    reject,
+  })
+}
+
+const play = (time?: number) => {
+  return new Promise((resolve, reject) => {
+    if (!canPlay.value) {
+      return reject(new Error("Player: Can't play"))
+    }
+    flushPreviouseActions()
+      .then(() => {
+        loading.value = true
+
+        if (time) {
+          audio.value.currentTime = time
+        }
+
+        const playPromise: Promise<any> = audio.value.play()
+
+        playPromise
+          .then(() => {
+            playerEventBus.emit(PlayerEvent.PLAY, { id: id.value })
+            resolve(true)
+          })
+          .catch(reject)
+          .finally(() => {
+            loading.value = false
+          })
+
+        pushToActionStack({ promise: playPromise, reject })
+      })
+      .catch(reject)
+  })
+}
+
+const pause = async () => {
+  return new Promise((resolve, reject) => {
+    if (!canPause.value) {
+      return reject(new Error("Player: Can't pause"))
+    }
+
+    flushPreviouseActions()
+      .then(() => {
+        audio.value.pause()
+        resolve(true)
+      })
+      .catch(reject)
+  })
+}
+
+const change = (time: number) => {
+  play(time)
+}
+
+const toggleMute = () => {
+  audio.value.muted = !muted.value
+}
+
+const goToEnd = () => {
+  if (!canStartPlaying.value) {
+    return
+  }
+  play(duration.value)
+}
+
+useEventListener(audio, 'canplaythrough', () => {
```

Comment:
How about change this event to 'canplay'?

That should give the feeling of quick loading and it willing be online with exez suggestion


https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canplay_event